### PR TITLE
python3Packages.polars: 0.17.11 -> 0.18.0 + unmark broken on aarch64-linux

### PR DESCRIPTION
**Pull request by kirillrdy**
###### Description of changes

https://github.com/pola-rs/polars/compare/py-0.17.11...py-0.18.0

cant upgrade to >0.18.0 because of dependency on rust 1.71

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 237495` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.polars</li>
    <li>python310Packages.polars.dist</li>
    <li>python311Packages.polars</li>
    <li>python311Packages.polars.dist</li>
  </ul>
</details>

**Comment by supermarin** [comment]
Hm, this is broken for me on x86_64-linux

File: pkgs/development/python-modules/polars/default.nix [hunk 1]
```diff
@@ -66,7 +64,6 @@ buildPythonPackage {
   # ];
 
   meta = with lib; {
-    broken = (stdenv.isLinux && stdenv.isAarch64);
```
Comment (supermarin) [inline pkgs/development/python-modules/polars/default.nix:69]: I believe you wanted `stdenv.isLinux && !stdenv.isAarch64`

Comment (kirillrdy) [reply pkgs/development/python-modules/polars/default.nix:69]: It did build on all platforms.  It's probably something from recent staging-next merge. I'll have a look at it today

Comment (supermarin) [reply pkgs/development/python-modules/polars/default.nix:69]: Thanks, appreciate that! looking into possibly bumping this further, current version is 0.18.13.

Comment (kirillrdy) [reply pkgs/development/python-modules/polars/default.nix:69]: Yes I was waiting for rust 1.71

**Comment by icyrockcom** [comment]
> Hm, this is broken for me on x86_64-linux

Interesting. I just pushed a 0.18.13 PR https://github.com/NixOS/nixpkgs/pull/248214, which does not change the `broken` attribute and it's definitely not broken for me:

```
$ python
Python 3.11.4 (main, Jun  6 2023, 22:16:46) [GCC 12.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.machine()
'x86_64'
>>> import polars
>>> polars.__version__
'0.18.13'
>>> 
```

This morning I was trying 0.18.0 as well and it worked fine, though I think that was from cache. I'm compiling it right now and will let you know if the results are different when it finishes (it takes a while...).

**Comment by icyrockcom** [comment]
@supermarin I can confirm compilation fails for me as well, with a couple of similar errors:

```
error[E0277]: the trait bound `T: SimdCast` is not satisfied
  --> /build/source/polars/polars-arrow/src/kernels/agg_mean.rs:25:26
   |
25 |         reduced += chunk.cast::<f64>();
   |                          ^^^^ the trait `SimdCast` is not implemented for `T`
   |
note: required by a bound in `std::simd::Simd::<T, N>::cast`
  --> /build/rustc-1.71.1-src/library/core/src/../../portable-simd/crates/core_simd/src/vector.rs:325:5
help: consider further restricting this bound
   |
17 |     T: NativeType + SimdElement + ToPrimitive + std::simd::SimdCast,
   |                                               +++++++++++++++++++++

error[E0277]: the trait bound `T: SimdCast` is not satisfied
  --> /build/source/polars/polars-arrow/src/kernels/agg_mean.rs:57:43
   |
57 |             let chunk = Simd::from(chunk).cast::<f64>();
   |                                           ^^^^ the trait `SimdCast` is not implemented for `T`
   |
note: required by a bound in `std::simd::Simd::<T, N>::cast`
  --> /build/rustc-1.71.1-src/library/core/src/../../portable-simd/crates/core_simd/src/vector.rs:325:5
help: consider further restricting this bound
   |
44 |     T: NativeType + SimdElement + ToPrimitive + IsFloat + std::simd::SimdCast,
   |                                                         +++++++++++++++++++++

For more information about this error, try `rustc --explain E0277`.
error: could not compile `polars-arrow` (lib) due to 2 previous errors
```

However, this is not due to polars itself not working on x64 - I use it daily on x64. :) 

It might be due to recent upgrade to rust 1.71.1 (it was 1.70.0 when @happysalada built it I think). The cached version should definitely work though.

So I guess you may want to temporarily downgrade your nixpkgs to a working version or you can try using my PR to get you 0.18.13. Or wait a couple of weeks, hopefully by then it will get approved and built by hydra. :)

**Comment by supermarin** [comment]
@icyrockcom whoa that was fast! went to build from your branch since I'm on flakes, seen it's already merged. Building now from [c27f08a](https://github.com/NixOS/nixpkgs/commit/c27f08aa6c12bc9a4ea45e43c1e29695e339b214), it's past the failure above, will report back if it fails elsewhere, if my build box doesn't melt before that

**Comment by supermarin** [comment]
Just for posterity: it compiled, confirming it works on x86_64-linux! Thanks @icyrockcom

**Pull request merged**